### ISSUE TITLE
Mat/Pat - CRM ticket - Thread 

### ISSUE THREAD

**Issue by fazil-ey** [bcgov/digital-journeys#1190]
Split from #1100 
The scope of this ticket is to format the the data being sent in the thread

*AC*
- [x] The CRM ticket should contain all form fields in the ticket details section  
    - [x] The text in the thread should be in a format that is easy to read (Sample Below)
    - [x] ensure leave end dates are included 

Fields that should be in the thread - 

- [x] Employee information and confirmation, Alternate contact information (if available),
- [x] if taking maternity leave - start date, end date, number of weeks, leave allowance selection,
- [x] if taking parental leave - Partners info and confirmation (if with partner in BC gov), parental leave type, number of week, childs birth date (if taking alone), start date, end date, leave allowance type, leave allowance dates, deferral date (if deferring), supervisor information

**Sample**
Click to enlarge
<img src="https://images.zenhubusercontent.com/62e986a3ff9c07df27cce462/173132cc-529b-4ff7-8eb9-4dad8821861d" width=50% length=50%/>

**Comment by Stella-Archer** [comment]
@iman-jamali-fw could you let us know where this ticket is at and if you have and estimate? thx

**Comment by iman-jamali-fw** [comment]
@Stella-Archer There should be no barrier to starting working on this ticket. I estimated it as 3, @bhumin-fw feel free to adjust it if needed.

**Comment by bhumin-fw** [comment]
@fazil-ey We need to identify fields that we want to display in thread. Please check attached image in the ticket for reference.
Let me know if you have any query here.

**Comment by fazil-ey** [comment]
@bhumin-fw it shoudl be all fields we show in review section

 Employee information and confirmation, Alternate contact information (if available), 
  - if taking maternity leave - start date, end date, number of weeks,  leave allowance selection, 
  - if taking parental leave - Partners info and confirmation (if with partner in BC gov), parental leave type, number of week,  childs birth date (if taking alone), start date, end date, leave allowance type, leave allowance dates, deferral date (if deferring), supervisor information

**Comment by bhumin-fw** [comment]
Ticket is ready to test on DEV & TEST. Can be test with #1124 as well.

**Comment by Stella-Archer** [comment]
* [ ] Employee information and confirmation
* [ ] Alternate contact information (if available),
Maternity leave 
* [ ] start date, end date,
* [ ] number of weeks,
* [ ] leave allowance selection,
Parental leave - 
* [ ] lPartners info and confirmation (if with partner in BC gov), 
* [ ] lparental leave type,
* [ ] number of week, 
* [ ] childs birth date (if taking alone), 
* [ ] start date, end date, 
* [ ] leave allowance type
* [ ] leave allowance dates, 
* [ ] deferral date (if deferring), 
* [ ] supervisor information

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbjhpIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--55aec1a37e10bc1426ff949493b4f0c2c7d86c55/image.png)

**Comment by Stella-Archer** [comment]
@fazil-ey  could you check to see if this is what I should be seeing in the CRM tix?
I'm not sure about:
- [ ] Mat leave allowance selection
- [ ] Pat leave dates
I'll do another test for partner info and deferring, and pat alone birthdate but really not sure if i'me seeing the above points

**Comment by bhumin-fw** [comment]
@fazil-ey @Stella-Archer 
Right now, we have covered only one page - "Review".
If fields/details are on the review page then those will be there on Thread as well.

Change applied till today's morning. If any change is made on the review page section later, that needs to be updated on Thread as well.

**Please let me know field/information which is not there on the review page but we want to display on CRM Thread.**

**Comment by fazil-ey** [comment]
@bhumin-fw  @Stella-Archer based on the screenshot, i think the fields you mentioned are missing
- the confirmation is missing for employee information
- Mat leave allowance selection is also missing
- parental leave allowance dates are also missing


i am not sure if the form submitted had partner info. We should also test the scenario of only parental leave.

**Comment by fazil-ey** [comment]
We are not getting CRM emails and submissions are not getting validated (status did not change to validated in submitted forms tab) . Submission done my Fazil with Stella and Ayush as supervisors, submissions done by Stella with herself as supervisor. 

@iman-jamali-fw  @bhumin-fw

**Comment by iman-jamali-fw** [comment]
@fazil-ey 
We deployed a fix to DEV and TEST that handles an edge case allowing employees (contractors with no employeesIds) to have CRM tickets. Please try again, it should be working again.

**Comment by fazil-ey** [comment]
@iman-jamali-fw  Test Aug 24 - got the emails! will test different scenarios and see if we are getting all fields in the CRM  now

**Comment by fazil-ey** [comment]
Hey @bhumin-fw - the thread looks good. However there is some missing data

- instead of _Standard, extended leave: Standard Leave Up To 35 Weeks_ can we say "Type of leave: Standard"
- Partner info is missing and "Does your partner work for BC Public Service" is also missing
- parental leave Allowance type, allowance dates are missing
- maternity leave allowance selection is missing 
email and form for comparison 
[emailCRM.pdf](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdHc0IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--f071343c816777e9f9cffe9f88ae8170674486d9/emailCRM.pdf)
[Digital Journeys.pdf](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdG80IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--9f84ec355b4c6dbbb014db80598bbaec1e7944bb/Digital%20Journeys.pdf)

For only parental leave
- When was your baby born or placed after adoption? is missing (also in review page)
- Are you a birthing parent? is also missing

**Comment by bhumin-fw** [comment]
@fazil-ey 
Let me add all missing details for thread here. I'll refer your last comment to update thread information. 
Let me know if anything else that needs to consider in thread details.

**Comment by ayushdamani** [comment]
- Tested in TEST environment.
- Submitted a mat/pat form.
- Validated the submission.
- Received the Mat/Pat CRM 

Acceptance criteria met:
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBc1IzIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--5505ada1875e4ee9dd7595eac897e4b7c267dcd1/image.png)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBc2QzIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--e91ce7f44f8906a6cb306fe7edb6bff2ab336979/image.png)

**Comment by fazil-ey** [comment]
@bhumin-fw  - mostly good. Minor changes are needed

1. This field is a duplicate now, since we have the option Leave Allowance in both parental and maternity leave
 DO YOU WANT TO RECEIVE TOP-UP ALLOWANCE?
Top Up Allowance: I Want To Receive My Top Up Allowance During My Leave After Acceptance From Ei

2. There is a missing space in "EmployeeNumber"
3. "No of weeks for standard leave" - Change to Number instead of No

**Comment by bhumin-fw** [comment]
@fazil-ey @ayushdamani 
Fix all 3 points from the recent comment.

**Comment by ayushdamani** [comment]
Tried to test this today:
1. Submitted a Mat/Pat form with myself as the supervisor.
2. Still could not validate my form submission.

Next step: I will request Meghan/Stella to submit a form with me as a supervisor to test this ticket.

**Comment by ayushdamani** [comment]
Tested and confirmed updated ✅
1. Meghan submitted a form with me as a supervisor
2. I validated the form
3. CRM ticket was generated
4. Checked the details against Fazil's comment

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBamVRIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--cdddaab1769ed972a0e570ff86333c794ac4a864/Screenshot%202023-09-12%20092805.png)

Thank you, @bhumin-fw
